mlflow/genai/optimize/types.py

```diff
@@ -118,3 +118,52 @@ class OptimizerOutput:
     optimizer_name: str
     final_eval_score: float | None = None
     initial_eval_score: float | None = None
+
+
+@experimental(version="3.5.0")
+@dataclass
+class EvaluationResultRecord:
+    """
+    An output type of `eval_fn` in the
```

Comment:
nit: An output type => The output type

Also is the docstring correct? looks like `mlflow.genai.optimize.BasePromptAdapter.optimize()` returns a different type.